RecoLocalTracker/SiStripZeroSuppression/plugins/SiStripBaselineComparator.cc

```diff
@@ -39,178 +42,134 @@
-       }
-       if(nMatched>1) h1_matchingMult_->Fill(clus->amplitudes().size(),nMatched);      
-     }
-   }
-
-      //clusters	  
-      /*    int nclust = 0;
-	  edmNew::DetSetVector<SiStripCluster>::const_iterator itClusters = clusters->begin();
-	  for ( ; itClusters != clusters->end(); ++itClusters ){
-		for ( edmNew::DetSet<SiStripCluster>::const_iterator clus =	itClusters->begin(); clus != itClusters->end(); ++clus){
-		  if(itClusters->id() == detId){
-		    int firststrip = clus->firstStrip();
-	            //std::cout << "Found cluster in detId " << detId << " " << firststrip << " " << clus->amplitudes().size() << " -----------------------------------------------" << std::endl;		
-     		    strip=0;
-		    for( auto itAmpl = clus->amplitudes().begin(); itAmpl != clus->amplitudes().end(); ++itAmpl){
-		      h1Clusters_->Fill(firststrip+strip, *itAmpl);
-		      ++strip;
-		    }
-		  }
-                  
-                  //cluster plots from here on
-                   
-                  if(ClusterDists == false){
-		    nclust++;
-
-     		    int strip2=0;
-                    double charge = 0;
-                    double mean = 0;
-                    double sigma = 0;
-		    for( auto itAmpl = clus->amplitudes().begin(); itAmpl != clus->amplitudes().end(); ++itAmpl){
-                      charge += *itAmpl;
-		      ++strip2;
-                      mean += strip2*(*itAmpl);
-		      sigma += strip2*strip2*(*itAmpl);
-		    }
-                    h1ClusterCharge_->Fill(charge);
-                    h1ClusterWidth_->Fill(strip2);
-                    mean = mean/charge;
-                    h1ClusterMean_->Fill(mean);
-                    sigma = TMath::Power((sigma/charge-mean*mean),0.5);
-                    h1ClusterSigma_->Fill(sigma);
-            }              
-	 }
-      }*/
+  edm::Handle<edmNew::DetSetVector<SiStripCluster> > clusters;
+  e.getByToken(srcClusters_,clusters);
+  edm::Handle<edmNew::DetSetVector<SiStripCluster> > clusters2;
+  e.getByToken(srcClusters2_,clusters2);
+
+  for (auto const& set : *clusters){
+    for (auto const& clus : set){ 
+      h1_nOldClusters_->Fill(clus.amplitudes().size(),1);
+      int nMatched = 0; 
+      int charge1 = 0;
+      for( auto itAmpl = clus.amplitudes().begin(); itAmpl != clus.amplitudes().end(); ++itAmpl) charge1 += *itAmpl;
+      std::vector< int > matchedWidths;      
+      std::vector< int > matchedCharges;      
+
+      //scan other set of clusters
+      for (auto const& set2 : *clusters2){
+        if(set.id() != set2.id()) continue;
+        for (auto const& clus2 : set2){ 
+          int charge2 = 0;
+          for( auto itAmpl = clus2.amplitudes().begin(); itAmpl != clus2.amplitudes().end(); ++itAmpl) charge2 += *itAmpl;
+          int strip=clus.firstStrip();
+          for( auto itAmpl = clus.amplitudes().begin(); itAmpl != clus.amplitudes().end(); ++itAmpl){
```

Comment:
for  range loop, or even a simple int loop from 0 to  ```clus.amplitudes().size()```.
There is no dependence on itAmpl in this loop.
I guess the loop is not even needed and can be replaced by ```if (clus2.firstStrip() >= clus.firstStrip() && clus2.firstStrip() < clus.firstStrip() + clus.amplitudes().size() )``` (and drop the ```break;```)